src/test/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisServiceTest.java

```diff
@@ -0,0 +1,368 @@
+        List<SensitivityFactor> aleaFactors;
+        if (mayContingencies == null) {
+            aleaFactors = List.of(new SensitivityFactor(MW_FUNC_TYPE, funcId, MW_VAR_TYPE, varId, false, ContingencyContext.all()));
+        } else {
+            aleaFactors = mayContingencies.stream().map(aleaId ->
+                    new SensitivityFactor(MW_FUNC_TYPE, funcId, MW_VAR_TYPE, varId, false,
+                        ContingencyContext.create(aleaId, ContingencyContextType.SPECIFIC)))
+                .collect(Collectors.toList());
+            aleaFactors.add(0, new SensitivityFactor(MW_FUNC_TYPE, funcId, MW_VAR_TYPE, varId, false, ContingencyContext.none()));
+        }
+        return aleaFactors;
+    }
+
+    private static SensitivityAnalysisInputData getDummyInputData() {
+        return SensitivityAnalysisInputData.builder()
+            .resultsThreshold(0.10)
+            .sensitivityInjectionsSets(List.of())
+            .sensitivityInjections(List.of())
+            .sensitivityHVDCs(List.of())
+            .sensitivityPSTs(List.of())
+            .sensitivityNodes(List.of())
+            .parameters(SensitivityAnalysisParameters.load())
+            .build();
+    }
+
+    @Test
+    public void test0() {
+        testBasic(true);
+        testBasic(false);
+    }
+
+    private void testBasic(boolean specific) {
+        List<String> aleaIds = List.of("a1", "a2", "a3");
+        final List<SensitivityAnalysisResult.SensitivityContingencyStatus> contingenciesStatuses = aleaIds.stream()
+            .map(aleaId -> new SensitivityAnalysisResult.SensitivityContingencyStatus(aleaId, SensitivityAnalysisResult.Status.SUCCESS))
+            .collect(Collectors.toList());
+
+        final List<SensitivityFactor> sensitivityFactors = Stream.of(
+                makeMWFactors("l1", "GEN", null), // factor index 0      ; for all()
+                makeMWFactors("l2", "GEN", List.of()), // factor index 1 ; for none()
+                makeMWFactors("l3", "LOAD", aleaIds)   // factor indices 2, 3, 4 & 5
+            ).flatMap(Collection::stream).collect(Collectors.toList());
+
+        final List<SensitivityValue> sensitivityValues = new ArrayList<>(List.of(
+            // l1 x GEN, before + a1, a2, a3 (implicit)
+            new SensitivityValue(0, -1, 500.1, 2.9),
+            new SensitivityValue(0, 0, 500.3, 2.7),
+            new SensitivityValue(0, 1, 500.4, 2.6),
+            new SensitivityValue(0, 2, 500.5, 2.5),
+            // l2 x GEN, just before
+            new SensitivityValue(1, -1, 500.2, 2.8),
+            // l3 x LOAD, before + a1, a2, a3 (explicit)
+            new SensitivityValue(2, -1, 500.9, 2.1),
+            new SensitivityValue(specific ? 3 : 2, 1, 500.6, 2.4),
+            new SensitivityValue(specific ? 4 : 2, 0, 500.7, 2.3),
+            new SensitivityValue(specific ? 5 : 2, 2, 500.8, 2.2)
+        ));
+        Collections.shuffle(sensitivityValues);
```

Comment:
Is this necessary ?

Reply:
Necessary ? Not entirely, but it shows order should not matter, as it challenges this risk.